lib/ansible/plugins/__init__.py

```diff
@@ -57,6 +59,17 @@ class AnsiblePlugin(ABC):
 
     def __init__(self):
         self._options = {}
+        self._hash = None
+
+    def _gen_signature(self):
```

Comment:
```suggestion
    def _gen_signature(self, variables=None):
```

Reply:
been going back and forth on this, if the options are not set .. we would need variables and task keys here

tempted to just force 'set options' if hash is not there